lib/hl7/datetime_components.rb

```diff
@@ -37,15 +40,17 @@ def ==(other)
     def validate!
       validate_field!(:year, -9999..9999)
       validate_field!(:month, 1..12)
-      validate_field!(:day, 1..Time.days_in_month(month, year))
+      validate_field!(:day, 1..Time.days_in_month(month, year)) unless day.nil?
       validate_field!(:hour, 0..23)
       validate_field!(:minute, 0..59)
-      validate_field!(:second, 0...60, :real?)
+      validate_field!(:second, 0..59)
+      validate_field!(:fraction, 0...1, :real?)
       validate_field!(:offset_seconds, -64800..64800)
     end
 
     def validate_field!(field, range, predicate = :integer?)
       value = public_send(field)
+      return if value.nil?
```

Comment:
aren't there certain things that we would expect not to be `nil`? Should we raise an error in that case?

Reply:
You're right. `year` must be non-nil.